src/hotspot/share/utilities/concurrentHashTable.hpp

```diff
@@ -192,9 +192,11 @@ class ConcurrentHashTable : public CHeapObj<F> {
   };
 
   // Used as default functor when no functor supplied for some methods.
+  // Note it only accepts the VALUE, and does not define methods with
+  // non-void VALUE returns. Doing so would require defining the neutral
+  // value for VALUE.
   struct NoOp {
     void operator()(VALUE*) {}
-    const VALUE& operator()() {}
     void operator()(bool, VALUE*) {}
   } noOp;
```

Comment:
I don't think the noOp object is useful.  I think it would be clearer to just use a constructed object of this type in the one place it's used, i.e. `NoOp()` or (if name is changed) `IgnoreValue()`.

Reply:
If we were using lambdas we could delete this class and just use `[](VALUE*) {}` in the one place it's being used.